go/cmd/vtbench/vtbench.go

```diff
@@ -86,22 +86,32 @@ import (
 */
 
 var (
-	// connection flags
-	host       = flag.String("host", "", "vtgate host(s) in the form 'host1,host2,...'")
-	port       = flag.Int("port", 0, "vtgate port")
-	unixSocket = flag.String("unix_socket", "", "vtgate unix socket")
-	protocol   = flag.String("protocol", "mysql", "client protocol, either mysql (default), grpc-vtgate, or grpc-vttablet")
-	user       = flag.String("user", "", "username to connect using mysql (password comes from the db-credentials-file)")
-	db         = flag.String("db", "", "db name to use when connecting / running the queries (e.g. @replica, keyspace, keyspace/shard etc)")
-
-	// test flags
-	deadline = flag.Duration("deadline", 5*time.Minute, "maximum duration for the test run (default 5 minutes)")
-	sql      = flag.String("sql", "", "sql statement to execute")
-	threads  = flag.Int("threads", 2, "number of parallel threads to run")
-	count    = flag.Int("count", 1000, "number of queries per thread")
+	host, unixSocket, user, db, sql string
+	port                            int
+	protocol                        = "mysql"
+	deadline                        = 5 * time.Minute
+	threads                         = 2
+	count                           = 1000
 )
 
+func initFlags() {
+	pflag.StringVar(&host, "host", "", "vtgate host(s) in the form 'host1,host2,...'")
+	pflag.IntVar(&port, "port", 0, "vtgate port")
+	pflag.StringVar(&unixSocket, "unix_socket", "", "vtgate unix socket")
+	pflag.StringVar(&protocol, "protocol", protocol, "client protocol, either mysql (default), grpc-vtgate, or grpc-vttablet")
+	pflag.StringVar(&user, "user", "", "username to connect using mysql (password comes from the db-credentials-file)")
+	pflag.StringVar(&db, "db", "", "db name to use when connecting / running the queries (e.g. @replica, keyspace, keyspace/shard etc)")
+
+	pflag.DurationVar(&deadline, "deadline", deadline, "maximum duration for the test run (default 5 minutes)")
+	pflag.StringVar(&sql, "sql", "", "sql statement to execute")
+	pflag.IntVar(&threads, "threads", threads, "number of parallel threads to run")
+	pflag.IntVar(&count, "count", count, "number of queries per thread")
+}
+
 func main() {
+	servenv.OnParse(func(fs *pflag.FlagSet) {
+		initFlags()
+	})
 	logger := logutil.NewConsoleLogger()
 	flag.CommandLine.SetOutput(logutil.NewLoggerWriter(logger))
```

Comment:
@ajm188, I was not sure whether this needs to be removed or not. I can see that https://github.com/vitessio/vitess/pull/11364 is getting rid of it. Let me know what I should follow :)

Reply:
well, it depends on what you want 😅 

if you want to keep the behavior, you want to something like:

```go
servenv.OnParse(func(fs *pflag.FlagSet) {
    fs.SetOutput(logutil.NewLoggerWriter(logger))
}
```